test/suites/metrics.sh

```diff
@@ -10,9 +10,13 @@ test_metrics() {
   # c1 metrics should show as the container is running
   lxc query "/1.0/metrics" | grep "name=\"c1\""
 
-  # c2 metrics should show the container as stopped
-  lxc query "/1.0/metrics" | grep "name=\"c2\""
-  lxc query "/1.0/metrics" | grep "name=\"c2\"" | grep "state=\"STOPPED\""
+  # c2 metrics should not exist as it's not running
+  ! lxc query "/1.0/metrics" | grep "name=\"c2\"" || false
+
+  # Check that we can get the count of existing instances.
+  lxc query /1.0/metrics | grep -xF 'lxd_instances{project="default",type="container"} 2'
+  # Ensure no lxd_instances reports no VM
+  ! lxc query /1.0/metrics | grep -F 'lxd_instances{project="default",type="virtual-machine"}' || false
```

Comment:
```suggestion
  # Ensure lxd_instances reports VM count properly (0)
  lxc query /1.0/metrics | grep -xF 'lxd_instances{project="default",type="virtual-machine"} 0'
```
I forgot the metric was always present :)

Reply:
fixed